applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.cpp

```diff
@@ -0,0 +1,854 @@
+            KRATOS_ERROR << "THERMAL_CONDUCTIVITY_SOLID_XY does not exist in the material properties or has an invalid value at element" << this->Id() << std::endl;
+        //
+        if (Prop.Has(THERMAL_CONDUCTIVITY_SOLID_YX) == false || Prop[THERMAL_CONDUCTIVITY_SOLID_YX] < 0.0)
+            KRATOS_ERROR << "THERMAL_CONDUCTIVITY_SOLID_YX does not exist in the material properties or has an invalid value at element" << this->Id() << std::endl;
+        //
+        if (Prop.Has(LONGITUDINAL_DISPERSIVITY) == false || Prop[LONGITUDINAL_DISPERSIVITY] < 0.0)
+            KRATOS_ERROR << "LONGITUDINAL_DISPERSIVITY does not exist in the material properties or has an invalid value at element" << this->Id() << std::endl;
+        //
+        if (Prop.Has(TRANSVERSE_DISPERSIVITY) == false || Prop[TRANSVERSE_DISPERSIVITY] < 0.0)
+            KRATOS_ERROR << "TRANSVERSE_DISPERSIVITY does not exist in the material properties or has an invalid value at element" << this->Id() << std::endl;
+        //
+        if (Prop.Has(SOLID_COMPRESSIBILITY) == false || Prop[SOLID_COMPRESSIBILITY] < 0.0)
+            KRATOS_ERROR << "SOLID_COMPRESSIBILITY does not exist in the material properties or has an invalid value at element" << this->Id() << std::endl;
+        //
+        if (TDim == 2) {
+            // If this is a 2D problem, nodes must be in XY plane
+            for (unsigned int i = 0; i < TNumNodes; ++i) {
+                if (Geom[i].Z() != 0.0)
+                    KRATOS_ERROR << " Node with non-zero Z coordinate found. Id: " << Geom[i].Id() << std::endl;
+            }
+        }
+        //
+        // KRATOS_INFO("1-TransientThermalElement::Check()") << this->Id() << std::endl;
+        KRATOS_CATCH("");
+        //
+        return 0;
+    }
+
+    // ============================================================================================
+    // ============================================================================================
+    template<unsigned int TDim, unsigned int TNumNodes>
+    void TransientThermalElement<TDim, TNumNodes>::InitializeSolutionStep(
+        const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY;
+        // KRATOS_INFO("0-TransientThermalElement::InitializeSolutionStep()") << this->Id() << std::endl;
+        //
+        if (!mIsInitialised) this->Initialize(rCurrentProcessInfo);
+        //
+        // KRATOS_INFO("1-TransientThermalElement::InitializeSolutionStep()") << std::endl;
+        KRATOS_CATCH("");
+    }
+
+    // ============================================================================================
+    // ============================================================================================
+    template< unsigned int TDim, unsigned int TNumNodes >
+    void TransientThermalElement<TDim, TNumNodes>::InitializeNonLinearIteration(
+        const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY;
+        // nothing
```

Comment:
if does nothing is better to remove the method

Reply:
4 methods are removed